### PR TITLE
std.math documentation update and @safe pure etc. unittests

### PR DESCRIPTION
**Pull request by burner**
* many small doc updates, a lot of examples 
* many unittests were marked safe, pure etc.

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -501,7 +502,7 @@ auto abs(Num)(Num z) @safe pure nothrow @nogc
     return hypot(z.re, z.im);
 }
 
-/** ditto */
+///
```
Comment (quickfur) [inline std/math.d]: Why this change? Shouldn't the overloads be ditto'ed together into a single doc entry?

**Comment by quickfur** [comment]
Other than that, LGTM. Thanks!

File: std/math.d (hunk 1 above)
Comment (burner) [inline std/math.d]: I will check the output

Comment (andralex) [inline std/math.d]: ping

**Comment by burner** [comment]
ditto again and some more doc unittests

**Comment by quickfur** [comment]
LGTM, thanks!

**Comment by andralex** [comment]
thx

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks
